### PR TITLE
Add spellcheck workflow

### PR DESCRIPTION
**Pull request by jjasghar**
We have a bunch of documentation, it wouldn't hurt to have spell check.

### CODE REVIEW

**Review by bjhargrave — CHANGES_REQUESTED** [review]
I think you also need to include the configuration files.
Screenshot 2024-04-16 at 16 17 00

**Review by lhawthorn — APPROVED** [review]
LGTM

**Review by bjhargrave — CHANGES_REQUESTED** [review]
Looking at the gha action, it requires a spellcheck.yml file which is not included in this PR.

For example: https://github.com/instructlab/instructlab/blob/main/.spellcheck.yml

I tested this PR in my fork and the spell check failed due to missing spellcheck.yml file.

```
Using pyspelling on configuration outlined in >spellcheck.yaml<
Checking files matching specified outlined in >spellcheck.yaml<
----------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/bin/pyspelling", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyspelling/__main__.py", line 30, in main
    return run(
           ^^^^
  File "/usr/local/lib/python3.12/site-packages/pyspelling/__main__.py", line 55, in run
    for results in spellcheck(
  File "/usr/local/lib/python3.12/site-packages/pyspelling/__init__.py", line 616, in spellcheck
    config = util.read_config(config_file)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyspelling/util/__init__.py", line 186, in read_config
    raise ValueError(
ValueError: Unable to find or load pyspelling configuration from spellcheck.yaml, for more details on configuration please read https://facelessuser.github.io/pyspelling/configuration/
```

File: .github/workflows/spellcheck.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+---
+name: Spellcheck
+
+on:
+  pull_request:
+    branches: [main]
+    paths:
+      - '**.md'
```
Comment (ckadner) [inline .github/workflows/spellcheck.yaml:8]:
```suggestion
      - .github/workflows/spellcheck.yaml
      - '**.md'
```

**Review by ckadner — DISMISSED** [review]
Nice!

**Comment by nathan-weinberg** [comment]
Very much in favor of this

**Comment by ckadner** [comment]
> > It doesn't seem to unless we want to add a specific dictionary.
> > https://github.com/marketplace/actions/github-spellcheck-action#configuration
> 
> Step 2 says you need to create a `spellcheck.yml` file.

https://github.com/instructlab/community/actions/runs/8835495973/job/24259773634?pr=162#step:4:19

```
  File "/usr/local/lib/python3.12/site-packages/pyspelling/__init__.py", line 616, in spellcheck
    config = util.read_config(config_file)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.12/site-packages/pyspelling/util/__init__.py", line 186, in read_config
    raise ValueError(
ValueError: Unable to find or load pyspelling configuration from spellcheck.yaml, for more details on configuration please read https://facelessuser.github.io/pyspelling/configuration/
Error: Files in repository contain spelling errors
```

**Comment by bjhargrave** [comment]
Also, this PR needs to fix any spelling errors that may exist so that the repo passes the spellcheck action.

**Comment by nathan-weinberg** [comment]
> Ok, I've got to say it: I don't know why everybody seems to love those spellcheckers. In my experience they are a pain because there generate too many false positives. We're not producing literature and there are just too many terms we use that spellcheckers don't know, and about every PR needs to include its additions to the custom dictionary. Now, feel free to ignore me, I've said my piece.

I think it makes a lot of sense given the doc-heavy nature of this repo, and I don't think the dictionary maintenence is going to be too hard once we get a lot of the base terms out of the way.

I've opened a PR for a Markdown Linter for this repo as well, they should pair well together: https://github.com/instructlab/community/pull/218

**Review by bjhargrave — APPROVED** [review]

**Review by nathan-weinberg — APPROVED** [review]

**Comment by nathan-weinberg** [comment]
@ckadner can you take another look and approve/merge if things look good to you?

**Pull request merged**